plugins/module_utils/utils.py

```diff
@@ -26,3 +28,174 @@ def generate_api_endpoint(path, **kwargs):
     # return full_url
 
     return path if not kwargs else "{0}?{1}".format(path, "&".join(["{0}={1}".format(key, value) for key, value in kwargs.items()]))
+
+
+def append_update_ops_data(ops, existing_data, update_path, changes=None, op="replace"):
```

Comment:
Should we expose replace and remove as two options, this way we do not have to specify the operation which is deduced from the passed in attribute. Furthermore this way we could process all operation in a single function call.

```suggestion
def append_update_ops_data(ops, existing_data, update_path, replace_data=None, remove_list=None):

if replace_data is None:
  replace_data = {}

if remove_data is None:
  remove_list = []
```